CONTRIBUTING.md ahead of move of repo to XRPLF (#4214)
XRPLF/rippled: CONTRIBUTING.md

# Contributing
The XRP Ledger has many and diverse stakeholders, and everyone deserves a chance to contribute meaningful changes to the code that runs the XRPL.
To contribute, please:
1. Fork the repository under your own user.

> 2. Create a new branch on which to write your changes. Please note that changes which alter the state or function of the ledger itself must be composed via and guarded using [Amendments](https://xrpl.org/amendments.html). Changes which are _read only_ i.e. RPC, or changes which are only refactors and maintain the existing behaviour do not need to be made through an Amendment.

Author: An amendment can change, for example, consensus itself (as in the `NegativeUNL` amendment.) I can change it if you want but it is reducing a more correct statement to a less correct statement.

> 2. Create a new branch on which to write your changes. Please note that changes which alter the state or function of the ledger itself must be composed via and guarded using [Amendments](https://xrpl.org/amendments.html). Changes which are _read only_ i.e. RPC, or changes which are only refactors and maintain the existing behaviour do not need to be made through an Amendment.

Reviewer: `NegativeUNL` introduced new transaction types, if it were implemented without on-ledger objects, it wouldn't have needed an amendment. I agree about changing state though, so a middle ground could be "changes which alter transaction processing and/or ledger state"?

10. Prefer readability over terseness where important logic is concerned.
11. Inline functions that are not used or are not likely to be used elsewhere in the codebase.
12. Use clear and self-explanatory names for functions, variables, structs and classes.
13. Use TitleCase for classes, structs and filenames, camelCase for function and variable names, lower case for namespaces and folders.
14. Provide as many comments as you feel that a competent programmer would need to understand what your code does.

# Maintainers
Maintainers are ecosystem participants with elevated access to the repository and are able to push new code, make decision on whether a release should be made, etc.

## Code Review
New contributors' PRs must be reviewed by at least two of the maintainers. Well established prior contributors can be reviewed by a single maintainer.

## Adding and Removing
New maintainers can be proposed by two existing maintainers, subject to a vote by a quorum of the existing maintainers. A minimum of 50% support and a 50% participation is required. In the event of a tie vote, the addition of the new maintainer will be rejected.

> Existing maintainers can resign or, be subject to a vote for removal at the behest of two existing maintainers. A minimum of 60% support and a 50% participation is required. The XRP Ledger Foundation will have the ability, for cause, to initiate the removal of an existing maintainer and such a request is not subject to a vote.

Reviewer 2: ```suggestion
Existing maintainers can resign, or be subject to a vote for removal at the behest of two existing maintainers. A minimum of 60% agreement and 50% participation are required. The XRP Ledger Foundation will have the ability, for cause, to remove an existing maintainer without a vote.
```
